onnxruntime/core/graph/graph_utils.cc

```diff
@@ -744,5 +745,44 @@ bool FindPath(const Node& node, bool is_input_edge, const std::vector<EdgeEndToM
   return true;
 }
 
+bool RemoveNodesWithOneOutputBottomUp(Graph& graph, const Node* start_node) {
+  if (start_node == nullptr) {
+    return false;
+  }
+  std::queue<const Node*> q;
```

Comment:
Use const Node& for start_node so you don't need this check.

Reply:
Updated.